Cleanup deprecated metrics

dgrisonnet opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind cleanup
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:

Remove the following deprecated metrics:
- node_collector_evictions_number
- scheduler_e2e_scheduling_duration_seconds

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
Remove the following deprecated metrics:
- node_collector_evictions_number replaced by node_collector_evictions_total
- scheduler_e2e_scheduling_duration_seconds replaced by scheduler_scheduling_attempt_duration_seconds
```

/cc @logicalhan

logicalhan reviewed (approved):
/lgtm
/approve

dgrisonnet commented:
/assign @k82cn @alculquicondor

alculquicondor commented on pkg/scheduler/metrics/metrics.go:
```diff
@@ -46,15 +46,6 @@ var (
-		&metrics.HistogramOpts{
-			Subsystem:         SchedulerSubsystem,
-			Name:              "e2e_scheduling_duration_seconds",
-			DeprecatedVersion: "1.23.0",
```
+1

we are 4 versions after.

alculquicondor commented on pkg/scheduler/metrics/metrics.go:
```diff
@@ -95,16 +86,6 @@ var (
-	SchedulerGoroutines = metrics.NewGaugeVec(
-		&metrics.GaugeOpts{
-			Subsystem:         SchedulerSubsystem,
-			DeprecatedVersion: "1.26.0",
```
I would keep this a couple of more versions (so remove in 1.29), just because this metric was quite old.

dgrisonnet replied:
I followed your call on that one and reverted the change.

Just as an fyi, in 1.27 this metric will be hidden by default meaning that it won't be exposed to end-users unless they specifically tell the scheduler to expose it.

alculquicondor replied:
oh is that a new change?

dgrisonnet committed:
Cleanup deprecated metrics

Remove the following deprecated metrics:
- node_collector_evictions_number
- scheduler_e2e_scheduling_duration_seconds

Signed-off-by: Damien Grisonnet <dgrisonn@redhat.com>

alculquicondor commented:
/approve
for scheduler. Thanks!

logicalhan reviewed (commented):
/lgtm

bowei commented:
/lgtm
/approve
